OIDC IdP API update for signed request object (#2800)
okta/okta-developer-docs: packages/@okta/vuepress-site/docs/reference/api/idps/index.md

    "algorithms": {
      "request": {
        "signature": {
          "algorithm": "HS256",
          "scope": "REQUEST"
        }
      },
      "response": {
        "signature": {
          "algorithm": "HS256",
          "scope": "ANY"
        }

>       }

Author: I included the `response` here because was originally here. But my code did not do anything about `response`, it will be ignored.

| Property    | Description                                                      | DataType                                          | Nullable | Readonly | MinLength |
| ----------- | ---------------------------------------------------------------- | ------------------------------------------------- | -------- | -------- | --------- |
|algorithms   | Settings for signing authorization requests                      | [OIDC Algorithms object](#oidc-algorithms-object) | TRUE    | FALSE    |

> |credentials  | Client authentication credentials for an [OAuth 2.0 Authorization Server (AS)](https://tools.ietf.org/html/rfc6749#section-2.3) | [Credentials object](#oauth-2-0-and-openid-connect-credentials-object) | FALSE | FALSE |   |

Reviewer: don't think we need to introduce the abbreviation for authorization server here
```suggestion
|credentials  | Client authentication credentials for an [OAuth 2.0 Authorization Server](https://tools.ietf.org/html/rfc6749#section-2.3) | [Credentials object](#oauth-2-0-and-openid-connect-credentials-object) | FALSE | FALSE |   |
```

Author: Hmm, looking at this again, I think I only added the row `algorithms` on top of the row `credentials`, I did not change or update the `credentials` row. In this case, would you still want me to remove the `(AS)`?

Author: https://developer.okta.com/docs/reference/api/idps/#oauth-2-0-protocol

Reviewer: if you don't mind and it doesn't cause any problems....it's not necessary

> | endpoints   | Endpoint settings for the OAuth 2.0 Authorization Server (AS)                                                                       | [OAuth 2.0 Endpoints object](#oauth-2-0-and-openid-connect-endpoints-object)  | TRUE  | TRUE  |   |

Reviewer: ```suggestion
| endpoints   | Endpoint settings for the OAuth 2.0 Authorization Server                                                                       | [OAuth 2.0 Endpoints object](#oauth-2-0-and-openid-connect-endpoints-object)  | TRUE  | TRUE  |   |
```

Author: Same with this. I'm happy to remove the `(AS)`, but I just want to double check as I didn't make any changes to this in my PR.

Reviewer: all good - please commit the change.

| Property    | Description                                                      | DataType                                          | Nullable | Readonly | MinLength |
| ----------- | ---------------------------------------------------------------- | ------------------------------------------------- | -------- | -------- | --------- |

> |algorithms   | Settings for signing authorization requests                      | [OIDC Algorithms object](#oidc-algorithms-object) <ApiLifecycle access="ea" /> | TRUE    | FALSE    |

Reviewer: @larsjohansen-okta - added one here

  }
}
```

> ##### OIDC Algorithms object

Reviewer 2: EA label here?

> The `OIDC` protocol supports `request` algorithm and verification settings.

Reviewer: ```suggestion
The `OIDC` protocol supports the `request` algorithm and verification settings.
```

Author: Thanks! Added `the`

> The `OIDC` protocol supports the `request` algorithm and verification settings.

Reviewer 2: EA label here?

| Property | Description                                                   | DataType                                                                 | Nullable | Readonly |
| -------- | ------------------------------------------------------------- | ------------------------------------------------------------------------ | -------- | ------   |
| request  | Algorithm settings used to sign an authorization request | [OIDC Request Algorithm object](#OIDC-request-algorithm-object)   | FALSE    | FALSE    |

```json
{
  "protocol": {
    "type": "OIDC",
    "algorithms": {
      "request": {
        "signature": {
          "algorithm": "HS256",
          "scope": "REQUEST"
        }
      }
    }
  }
}
```

> ###### OIDC Request Algorithm object

Reviewer 2: EA label here?

> Algorithm settings for signing authorization requests sent to the IdP:

Reviewer 2: EA label here?

| Property  | Description                                                 | DataType                                                                                   | Nullable | Readonly |
| --------- | ----------------------------------------------------------- | ------------------------------------------------------------------------------------------ | -------- | -------- |
| signature | Algorithm settings used to sign an authorization request | [OIDC Signature Algorithm object](#oidc-request-signature-algorithm-object) | FALSE    | FALSE    |

```json
{
  "protocol": {
    "type": "OIDC",
    "algorithms": {
      "request": {
        "signature": {
          "algorithm": "HS256",
          "scope": "REQUEST"
        }
      }
    }
  }
}
```

> ###### OIDC request Signature Algorithm object

Reviewer 2: EA label here?

> HMAC Signature Algorithm settings for signing authorization requests sent to the IdP:

Reviewer 2: EA label here?

| Property    | Description                                                                        | DataType             | Nullable | Readonly |
| ----------- | ---------------------------------------------------------------------------------- | -------------------- | -------- | -------- |
| algorithm   | The HMAC Signature Algorithm used when signing an authorization request  | `HS256`, `HS384` or `HS512` | FALSE    | FALSE    |

> | scope       | Specifies whether to digitally sign authorization request to the IdP | `REQUEST` or `NONE`  | FALSE    | FALSE    |

Reviewer: ```suggestion
| scope       | Specifies whether to digitally sign an authorization request to the IdP | `REQUEST` or `NONE`  | FALSE    | FALSE    |
```

> > **Note:** The `algorithm` property is ignored when disabling request signatures (`scope` set as `NONE`).

Reviewer: ```suggestion
> **Note:** The `algorithm` property is ignored when you disable request signatures (`scope` set as `NONE`).
```

Author: ok, I updated this. I'm also updating line 4970 to make it consistent
